other/upgrade.php

```diff
@@ -2725,6 +2729,27 @@ function cmdStep0()
 		require_once($modSettings['theme_dir'] . '/languages/Install.' . $upcontext['language'] . '.php');
 	}
 
+	// Do we need to add this setting?
+	$need_settings_update = empty($modSettings['custom_avatar_dir']);
+
+	$custom_av_dir = !empty($modSettings['custom_avatar_dir']) ? $modSettings['custom_avatar_dir'] : $GLOBALS['boarddir'] . '/custom_avatar';
+	$custom_av_url = !empty($modSettings['custom_avatar_url']) ? $modSettings['custom_avatar_url'] : $boardurl . '/custom_avatar';
```

Comment:
$boarddir is global here, don't use $GLOBALS

Reply:
I can make that change, but just so you're aware: $GLOBALS is used throughout.  I wasn't seeking to do a rewrite, just address the bug, in a fashion consistent with the existing code.

As noted in the comments, cmdstep0 is the CLI equivalent of WelcomeLogin(), and this code was cloned from there.

Reply:
In that case leave it.  Should address in the future.

Reply:
I had to bring the error rpt in line with the cli errors anyway, so a couple minor changes.  I think this is better.